src/openforms/contrib/haal_centraal/models.py

```diff
@@ -40,8 +41,62 @@ class HaalCentraalConfig(SingletonModel):
         default=BRPVersions.v13,  # TODO: should be change the default to v2?
         help_text=_("The API version provided by the selected service."),
     )
+    default_x_doelbinding_header = models.CharField(
+        _("x-doelbinding header"),
+        max_length=255,
+        blank=True,
+        help_text=_(
+            "The default 'x-doelbinding' header to be used when requesting the BRP API, "
+            "in case it is not available on the current form."
+        ),
+    )
+    default_x_verwerking_header = models.CharField(
+        _("x-verwerking header"),
+        max_length=242,
+        blank=True,
+        validators=[validate_verwerking_header],
+        help_text=_(
+            "The default 'x-verwerking' header to be used when requesting the BRP API, "
+            "in case it is not available on the current form."
+        ),
+    )
```

Comment:
> I don't expect other API's in the future to use the same headers

How was I suppose to know this? Anyway I think this relates to what was discussed a while ago on the fact that we are missing a lot of context on how all these APIs/services/movement/initiatives relate together

Reply:
> that's the only API where they are currently used, and I don't expect other API's in the future to use the same headers, looking at how things have been designed traditionally...

I hope this will be the case as if a new haal centraal service needs these headers we will need to change the field names. For reference, this is how `open-inwoner` [handles it](https://github.com/maykinmedia/open-inwoner/blob/develop/src/open_inwoner/haalcentraal/models.py); but I'm not sure if they are doing it correctly as well

Reply:
> How was I suppose to know this? Anyway I think this relates to what was discussed a while ago on the fact that we are missing a lot of context on how all these APIs/services/movement/initiatives relate together

You weren't - precisely because of the reasons you mention! After seeing this, I started putting more information on Taiga about this and started a discussion on Slack to try and provide more education. Feel free to extend this Taiga documentation with additional information, I've only done the bare minimum to get it started so far.

> I hope this will be the case as if a new haal centraal service needs these headers we will need to change the field names. For reference, this is how open-inwoner [handles it](https://github.com/maykinmedia/open-inwoner/blob/develop/src/open_inwoner/haalcentraal/models.py); but I'm not sure if they are doing it correctly as well

If this becomes the case, then we change the field names, but we can at least trace it back to a request for change. It's easier to make things more generic from specific compared to the other way around.

I can't speak about Open Inwoner - it's not a project I'm directly involved with but I know that I'm one of the strictest people for reviewing etc. so I'm sure I'll have _opinions_ if I were to look into it in more detail :stuck_out_tongue:

Reply:
> You weren't - precisely because of the reasons you mention! After seeing this, I started putting more information on Taiga about this and started a discussion on Slack to try and provide more education. Feel free to extend this Taiga documentation with additional information, I've only done the bare minimum to get it started so far.

I'll see in the future if I can make some kind of diagram or image that explains the relation between these, and also how it relates with the open forms solo models; I think that could be valuable